pkg/controller/flinkapplication/flink_state_machine_test.go

```diff
@@ -81,6 +81,132 @@ func TestHandleStartingClusterStarting(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestHandleNewOrCreateWithSavepointDisabled(t *testing.T) {
```

Comment:
This is great, but can you also add an integration test? Given this is a completely different flow, would be good to fully test out the successful and failure paths.

Reply:
done!